### PR TITLE
Only start reactions when the minerals are available in the current r…

### PR DESCRIPTION
**Pull request by ahmedcharles**

### CODE REVIEW

File: src/resources/Abathur.ts [hunk 1]
```diff
@@ -149,33 +149,6 @@ export class Abathur {
 		return this._globalAssets;
 	}
 
-	private canReceiveBasicMineralsForReaction(mineralQuantities: { [resourceType: string]: number },
```
Comment (bencbartlett) [inline src/resources/Abathur.ts]: why remove these two methods?

**Comment by bencbartlett** [comment]
I'd like to understand the reasoning behind these changes. If you only allow reactions to be started once all minerals are available, then this removes the ability of a room to actively request reactants from other rooms, which would decrease global mineral production. Have you tested these changes on public servers / been logging lab uptime in the grafana dashboard?

**Comment by ahmedcharles** [comment]
I observed the system getting stuck on trying to do a reaction when it was impossible for it to finish that reaction. I.e. it didn't request it from another room and it didn't buy it. I don't understand how it's supposed to work and it's not documented, so I removed the functionality that appeared to be broken.

I still think it's broken that it endlessly loops on a single reaction and never proceeds to the next reaction, so how is it supposed to work and what changes are required to stop it from looping when it's stuck?

File: src/resources/Abathur.ts (hunk 1 above)
Comment (ahmedcharles) [reply src/resources/Abathur.ts]: Both of these functions assume that something is going to either send or buy minerals for this room. I don't see how that happens and there's no docs that say where it happens and I observed it not happening, so I assumed they simply didn't work and removed them. If you can point out where the code is that does these things, I'll add comments to point it out and investigate why these returned true while the corresponding systems didn't actually work.

**Comment by ahmedcharles** [comment]
I think #110 fixes some of this but I'm not sure it works completely yet.

**Comment by zGeneral** [comment]
FYI
I tried it, it works. thanks
zGeneral

**Comment by bencbartlett** [comment]
@ahmedcharles Now that I have merged #110, is this still an issue?

**Comment by ahmedcharles** [comment]
I'm not sure. I'll need to test it, but I'll close this anyways and open a new issue/pr if I find something.

**Pull request closed without merging**